entrypoint.sh

```diff
@@ -26,36 +26,97 @@ echo "Starting: ACTION=$ACTION with DRY_RUN=$DRY_RUN"
 
 # Terraform output options
 export TF_CLI_ARGS=${TF_CLI_ARGS:-"-no-color"}
-TERRAFORM_OUT_DIR="./module"
-TERRAFORM_CMD="terraform -chdir=$TERRAFORM_OUT_DIR"
-TERRAFORM_VARS="-var-file=tfvars.json"
-OUTPUTS_FILE="${TERRAFORM_OUT_DIR}/outputs.json"
 
-# Generate module configuration
-# generate-tf-config is a script in the final module.
-generate-tf-config
+# The base terraform configuration to run
+export TERRAFORM_MODULE_DIR=${TERRAFORM_MODULE_DIR:-"./module"}
 
-$TERRAFORM_CMD init
+# Terraform will take the module path as a working directory
+export TERRAFORM_CMD="terraform -chdir=$TERRAFORM_MODULE_DIR"
 
-if [[ $ACTION == "Apply" ]]; then
-    if [[ $DRY_RUN == "True" ]]; then
-        $TERRAFORM_CMD plan -out=plan $TERRAFORM_VARS
-        if [ -f "validate_plan.py" ]; then
-            $TERRAFORM_CMD show -json $TERRAFORM_OUT_DIR/plan > $TERRAFORM_OUT_DIR/plan.json
-            python3 validate_plan.py "$TERRAFORM_OUT_DIR"/plan.json
-        fi
-    elif [[ $DRY_RUN == "False" ]]; then
-        $TERRAFORM_CMD apply -auto-approve $TERRAFORM_VARS
-        $TERRAFORM_CMD output -json > $OUTPUTS_FILE
-        if [ -f "post_checks.py" ]; then
-            python3 post_checks.py $OUTPUTS_FILE
+# the vars file path within the module directory
+export TERRAFORM_VARS="-var-file=tfvars.json"
+
+export PLAN_FILE="${APP}/plan"
+export PLAN_FILE_JSON="${APP}/plan.json"
+export OUTPUTS_FILE="${APP}/output.json"
```

Comment:
Please don't store these temporary files in `$APP`; this directory may be owned by root and not writeable.

Reply:
Moved to a temporary directory.